src/Interpreters/AsynchronousMetrics.h

```diff
@@ -55,8 +63,8 @@ class AsynchronousMetrics
 private:
     Context & global_context;
     const std::chrono::seconds update_period;
-    const std::vector<ProtocolServerAdapter> & servers_to_start_before_tables;
-    const std::vector<ProtocolServerAdapter> & servers;
+    std::shared_ptr<std::vector<ProtocolServerAdapter>> servers_to_start_before_tables{nullptr};
```

Comment:
Why shared_ptr?

Reply:
Because to construct this class with old behaviour I can't write `std::make_unique<AsynchronousMetrics>(*context, 60, {}, {})`, because a type of the latter arguments won't be deduced. Also I can't write `std::make_unique<AsynchronousMetrics>(*context, 60, std::vector<ProtocolServerAdapter>(), std::vector<ProtocolServerAdapter>())`, because ProtocolServerAdapter is an incompelete type in a point where I want to construct this class and I don't want to include Server/ProtocolServerAdapter.h there. Also `std::optional` doesn't work with references. Honestly saying, I am afraid of this construction because of probable dangling references. That's why I use shared_ptr. I would be happy if I was told how to do it right.